backtrace-library/src/main/cpp/backtrace-native.cpp

```diff
@@ -116,80 +158,119 @@ namespace /* anonymous */
 }
 
 extern "C" {
-    void Crash() {
-        *(volatile int *) 0 = 0;
-    }
-
-    void DumpWithoutCrash() {
-        crashpad::NativeCPUContext context;
-        crashpad::CaptureContext(&context);
-        client->DumpWithoutCrash(&context);
-    }
-
-    JNIEXPORT void JNICALL Java_backtraceio_library_base_BacktraceBase_crash(
-            JNIEnv *env,
-            jobject /* this */) {
-        Crash();
-    }
-
-    bool Initialize(jstring url,
-                    jstring database_path,
-                    jstring handler_path,
-                    jobjectArray attributeKeys,
-                    jobjectArray attributeValues) {
-        static std::once_flag initialize_flag;
-
-        std::call_once(initialize_flag, [&]{
-            initialized = InitializeImpl(url,
-                database_path, handler_path,
-                attributeKeys, attributeValues);
-        });
-        return initialized;
-    }
+void Crash() {
+    *(volatile int *) 0 = 0;
+}
 
-    JNIEXPORT jboolean JNICALL
-    Java_backtraceio_library_BacktraceDatabase_initialize(JNIEnv *env,
-                                                          jobject thiz,
-                                                          jstring url,
-                                                          jstring database_path,
-                                                          jstring handler_path,
-                                                          jobjectArray attributeKeys,
-                                                          jobjectArray attributeValues) {
-        return Initialize(url, database_path, handler_path, attributeKeys, attributeValues);
-    }
+void DumpWithoutCrash(jstring message) {
+    crashpad::NativeCPUContext context;
+    crashpad::CaptureContext(&context);
 
+    // set dump message for single report
+    crashpad::SimpleStringDictionary *annotations = NULL;
+    bool did_attach = false;
 
```

Comment:
Can we just call `AddAttribute` here as well to add the custom attribute `error.message` = `myMessage`? Why does it need to add the attribute in this custom way?

Reply:
// user can't override error.message - exception message that Crashpad/crash-reporting tool
 // will set to tell user about error message. This code will set error.message only for single
 // report and after creating a dump, method will clean up this attribute.

We still need to later opearte on annotations object, so this is the reason why I prefered to have this code twice.

Reply:
Ah okay I think I understand. This will set a one-time attribute "error.message" just for this specific dump called by `DumpWithoutCrash`. In contrast to the `AddAttribute` function which will add attributes for all subsequent dumps.